pkg/frontend/transport/handler.go

```diff
@@ -302,9 +315,25 @@ func (f *Handler) reportQueryStats(r *http.Request, queryString url.Values, quer
 	level.Info(util_log.WithContext(r.Context(), f.log)).Log(logMessage...)
 }
 
-func formatQueryString(queryString url.Values) (fields []interface{}) {
+// formatQueryString prefers printing start, end, and step from details if they are not nil.
+func formatQueryString(details *querymiddleware.QueryDetails, queryString url.Values) (fields []interface{}) {
 	for k, v := range queryString {
-		fields = append(fields, fmt.Sprintf("param_%s", k), strings.Join(v, ","))
+		var formattedValue string
+		if details != nil && (k == "start" || k == "end" || k == "step") {
+			var millis int64
+			switch k {
+			case "start":
+				millis = details.Start.UnixMilli()
+			case "end":
+				millis = details.End.UnixMilli()
```

Comment:
What if we emitted these in RFC 3339 format? Then they'd be human and machine readable.